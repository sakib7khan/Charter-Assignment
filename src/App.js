// App.js

import React, { useState } from 'react';
import CustomerList from './components/CustomerList/customerList';
import CustomerRewards from './components/CustomerRewards/customerRewards';
import FilterBar from './components/FilterBar/filterBar';
import TransactionTable from './components/TransactionTable/transactionTable';
import { Container, Title } from './styles/stylesComponents';

const App = () => {
  const [selectedCustomer, setSelectedCustomer] = useState('');
  const [selectedCustomerName, setSelectedCustomerName] = useState('');
  const [selectedMonth, setSelectedMonth] = useState('');
  const [selectedYear, setSelectedYear] = useState('2025');

  const handleSelectCustomer = (id, name) => {
    setSelectedCustomer(id);
    setSelectedCustomerName(name);
  };

  return (
    <Container>
      <Title>🏆 Rewards Tracker</Title>

      <div
        style={{
          display: 'flex',
          justifyContent: 'space-between',
          flexWrap: 'wrap',
          marginBottom: '24px',
        }}
      >
        <div style={{ flex: '1 1 40%' }}>
          <CustomerList onSelectCustomer={handleSelectCustomer} />
        </div>
        <div style={{ flex: '1 1 40%', textAlign: 'right' }}>
          <FilterBar
            selectedMonth={selectedMonth}
            selectedYear={selectedYear}
            onMonthChange={setSelectedMonth}
            onYearChange={setSelectedYear}
          />
        </div>
      </div>

      {selectedCustomer && (
        <>
          {/* 👇 Transactions table will come first */}
          <TransactionTable
            customerId={selectedCustomer}
            customerName={selectedCustomerName}
            selectedMonth={selectedMonth}
            selectedYear={selectedYear}
          />

          {/* 👇 Rewards Summary comes after transactions */}
          <CustomerRewards
            customerId={selectedCustomer}
            selectedMonth={selectedMonth}
            selectedYear={selectedYear}
          />
        </>
      )}
    </Container>
  );
};

export default App;
