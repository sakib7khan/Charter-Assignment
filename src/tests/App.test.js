import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';
import '@testing-library/jest-dom';

test('App component snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
