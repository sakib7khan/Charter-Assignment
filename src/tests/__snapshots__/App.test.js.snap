// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot 1`] = `
<div>
  <div
    class="sc-Qotzb fbOokM"
  >
    <h1
      class="sc-fYsHOw gOtxjZ"
    >
      🏆 Rewards Tracker
    </h1>
    <div
      style="display: flex; justify-content: space-between; flex-wrap: wrap; margin-bottom: 24px;"
    >
      <div
        style="flex: 1 1 40%;"
      >
        <div
          class="sc-iQQCXo bezgzY"
        >
          Loading customers...
        </div>
      </div>
      <div
        style="flex: 1 1 40%; text-align: right;"
      >
        <div
          class="sc-iQQCXo bezgzY"
        >
          Loading filters...
        </div>
      </div>
    </div>
  </div>
</div>
`;
